sophia/src/dataset/_traits.rs

```diff
@@ -714,10 +699,11 @@ pub trait MutableDataset: for<'x> Dataset<'x> {
                     q.g().map(BoxTerm::from),
                 )
             })
-            .collect::<std::result::Result<_, _>>()
+            .collect::<std::result::Result<Vec<_>, _>>()
             .map_err(Into::into)?;
         let mut to_remove = to_remove.into_iter().as_quad_source();
-        self.remove_all(&mut to_remove)?;
+        self.remove_all(&mut to_remove)
+            .map_err(|err| err.into_sink_err())?;
```

Comment:
according to my suggestion on `StreamError`, this would become
```suggestion
            .map_err(|err| err.unwrap_into())?;
```
I'm guessing the correct type will be inferred.